src/internal/util/rx.ts

```diff
@@ -0,0 +1,113 @@
+import { from } from '../observable/from';
+import { Observable } from '../Observable';
+import { ObservableInput, UnaryFunction } from '../types';
+import { pipeFromArray } from './pipe';
+
+export function rx<T, A>(source: ObservableInput<A>): Observable<A>;
+export function rx<T, A, B>(source: ObservableInput<A>, fn2: UnaryFunction<Observable<A>, B>): B;
+export function rx<T, A, B, C>(source: ObservableInput<A>, fn2: UnaryFunction<Observable<A>, B>, fn3: UnaryFunction<B, C>): C;
+export function rx<T, A, B, C, D>(
+  source: ObservableInput<A>,
+  fn2: UnaryFunction<Observable<A>, B>,
+  fn3: UnaryFunction<B, C>,
+  fn4: UnaryFunction<C, D>
+): D;
+export function rx<T, A, B, C, D, E>(
+  source: ObservableInput<A>,
+  fn2: UnaryFunction<Observable<A>, B>,
+  fn3: UnaryFunction<B, C>,
+  fn4: UnaryFunction<C, D>,
+  fn5: UnaryFunction<D, E>
+): E;
+export function rx<T, A, B, C, D, E, F>(
+  source: ObservableInput<A>,
+  fn2: UnaryFunction<Observable<A>, B>,
+  fn3: UnaryFunction<B, C>,
+  fn4: UnaryFunction<C, D>,
+  fn5: UnaryFunction<D, E>,
+  fn6: UnaryFunction<E, F>
+): F;
+export function rx<T, A, B, C, D, E, F, G>(
+  source: ObservableInput<A>,
+  fn2: UnaryFunction<Observable<A>, B>,
+  fn3: UnaryFunction<B, C>,
+  fn4: UnaryFunction<C, D>,
+  fn5: UnaryFunction<D, E>,
+  fn6: UnaryFunction<E, F>,
+  fn7: UnaryFunction<F, G>
+): G;
+export function rx<T, A, B, C, D, E, F, G, H>(
+  source: ObservableInput<A>,
+  fn2: UnaryFunction<Observable<A>, B>,
+  fn3: UnaryFunction<B, C>,
+  fn4: UnaryFunction<C, D>,
+  fn5: UnaryFunction<D, E>,
+  fn6: UnaryFunction<E, F>,
+  fn7: UnaryFunction<F, G>,
+  fn8: UnaryFunction<G, H>
+): H;
+export function rx<T, A, B, C, D, E, F, G, H, I>(
+  source: ObservableInput<A>,
+  fn2: UnaryFunction<Observable<A>, B>,
+  fn3: UnaryFunction<B, C>,
+  fn4: UnaryFunction<C, D>,
+  fn5: UnaryFunction<D, E>,
+  fn6: UnaryFunction<E, F>,
+  fn7: UnaryFunction<F, G>,
+  fn8: UnaryFunction<G, H>,
+  fn9: UnaryFunction<H, I>
+): I;
+export function rx<T, A, B, C, D, E, F, G, H, I>(
+  source: ObservableInput<A>,
+  fn2: UnaryFunction<Observable<A>, B>,
+  fn3: UnaryFunction<B, C>,
+  fn4: UnaryFunction<C, D>,
+  fn5: UnaryFunction<D, E>,
+  fn6: UnaryFunction<E, F>,
+  fn7: UnaryFunction<F, G>,
+  fn8: UnaryFunction<G, H>,
+  fn9: UnaryFunction<H, I>,
+  ...fns: UnaryFunction<any, any>[]
```

Comment:
Same question as with `pipe`: Can we replace the `any` usage here with `unknown`?